custom_code/hover_menu.js

```diff
@@ -0,0 +1,195 @@
+/**
+This is a script to allow cobrowsing for site navigation bar created using the "Sky Mega Menu"
+solution (http://voky.com.ua/showcase/sky-mega-menu/) in which dropdowns open and close based
+on CSS rules using the `:hover` pseudo-class.
+
+Pseudo-classes can not triggered programmatically - it has to be an actual user interaction.
+This means we cannot mirror the act of an user hovering a menu element to the other
+cobrowsing participant.
+
+This can be solved by adding an additional method (next to CSS) for opening and closing
+menu items. Namely, we can change menu items' styles based on user's mouse movements.
+This works because element style changes are caught by cobrowser and sent to the operator.
+
+Beware that this script is dependent on the structure of the menu and has been written
+solely based on the structure of a certain menu.
+**/
+
+// Helper function to query from direct children.
+var queryDirectChildren = function (parent, selector) {
+  if (!parent.children) return [];
+
+  var result = [];
+
+  for (var i = 0; i < parent.children.length; i++) {
+    var child = parent.children[i];
+    // `msMatchesSelector` is required for IE
+    var matches = child.matches || child.msMatchesSelector || undefined;
+    if (matches && matches.call(child, selector)) {
+      result.push(child);
+    }
+  }
+
+  return result;
+}
+
+// Function that styles for a sub-menu for it to be visible
+var expandMenu = function(subMenuContainerStyle, level, phoneMediaQueryList, right) {
+  subMenuContainerStyle.transform = 'scale(1, 1)';
+  subMenuContainerStyle.opacity = '1.0';
+
+  // Next steps are needed for setting sub-menu location,
+  // which depends on screen size, menu level and whether top menu is on the right
+
+
+  // First check if we're in phone view
+  if (phoneMediaQueryList.matches) {
+    if (level === 2) {
+      subMenuContainerStyle.left = '41px';
+    } else {
+      subMenuContainerStyle.left = '0';
+    }
+    // In phone view, level 1 menu items fill the whole space under menu bar
```

Comment:
Putting this check as a separate early return would IMO improve readability.

Reply:
10c5d262e7287ac34aa8f47ee19799eca8a75330